src/lib389/lib389/replica.py

```diff
@@ -1829,8 +1828,8 @@ def restore_changelog(self, replica_root, log=None):
             if not replica.task_finished():
                 raise ValueError("The changelog import task (LDIF2CL) did not complete in time")
         elif changelog_ldif_done:
-            ldif_done_file = os.path.join(cl_dir, changelog_ldif_done[0])
-            ldif_file = os.path.join(cl_dir, f"{replica_name}_cl.ldif")
+            ldif_done_file = os.path.join(ldif_dir, changelog_ldif_done[0])
+            ldif_file = os.path.join(ldif_dir, f"{replica_name}_cl.ldif")
             ldif_file_exists = os.path.exists(ldif_file)
```

Comment:
This is wrong,  The changelog export is not actually an ldif file, but contains partial LDAP modify statements used by replication.  It is supposed to go into the changelog directory, and not the server's LDIF directory

Reply:
Also, I am working on a 1.4.3 specific bug with dump-changelog

https://github.com/389ds/389-ds-base/issues/5202

Maybe hold off on this until I get my ticket fixed?

Reply:
I rechecked this on `dev` branch.

As I can see actually it should be `ldifdir` (not `cldir`) because the underlying `replica_execute_ldif2cl_task` looks for `ldif` file in `cl5GetLdifDir` => `slapi_ch_smprintf("%s/../ldif",dbdir)`.

But the code we discussed about is unreachable by from CLI.
Please see attached testcases for export/import changelog.

[export_import_changelog.testcases.txt](https://github.com/389ds/389-ds-base/files/8304546/export_import_changelog.testcases.txt)

In short, the following scenario should work IMHO, but it does not:
```
- to-ldif + preserve:default
# export
[root@replica1 /]# dsconf IPA-TEST replication export-changelog to-ldif -r 'dc=ipa,dc=test' -o cl_ipa-test.ldif --preserve-ldif-done

[root@replica1 /]# ls -1 /var/lib/dirsrv/slapd-IPA-TEST/ldif/
2e6c9b8c-a6af11ec-9da6a5b7-6375935c_cl.ldif.done

[root@replica1 /]# ls cl_ipa-test.ldif
cl_ipa-test.ldif

# import
[root@replica1 /]# dsconf IPA-TEST replication import-changelog default -r 'dc=ipa,dc=test'
Error: The default LDAP file "2e6c9b8c-a6af11ec-9da6a5b7-6375935c_cl.ldif" does not exist

# result: expected: OK, actual FAIL
```

`lib389/cli_conf/replication.py::restore_cl_def_ldif` doesn't respect `ldif.done` and if it is fixed then the discussed code can be checked and it works as expected.

Sorry, if I understand something wrongly.

Reply:
Starting in 1.4.4 we moved the replication change log into each backend, instead of it being a separate backend (each suffix now has its own changelog).  So that's why in newer branches it is under the ldif directory, but in 1.4.3 it is _not_.  It is directly placed in the changelog database directory (cl_dir). 

I thought on master branch the changelog export/import was refined and is working.  Are you saying it's not?

Reply:
Exactly, `ldif_dir` works just fine on `master` (note: this PR is for `master`).

Reply:
Oh... haha  My bad, I thought this was for 1.4.3.  Let me look this over one more time...